bin/tictactoe

```diff
@@ -1,3 +1,46 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+def start
+  input = ' '
+    puts "Welcome to Console Tic-Tac-Toe!\n
+    Please make a selection from bellow using your keybord:\n
+        Changes your mind? Please type 'exit'\n
+        Enter 1 to play vs. Computer.\n
+        Enter 2 to play vs. a buddy\n
+        Enter 3 for AI vs. AI\n"
+
+    input = gets.strip
+
+    case input
+    when '1'
+      single_player
+    when '2'
+      Game.new(Players::Human.new('X'), Players::Human.new('O'), Board.new).play
+    when '3'
+      Game.new(Players::Computer.new('X'), Players::Computer.new('O'), Board.new).play
+    end
+  end
+
+def single_player
+  puts "Would you like to go first\n Y/N"
+  if gets.strip == 'y'
+    Game.new(Players::Human.new('X'), Players::Computer.new('O'), Board.new).play
+  else
+    Game.new(Players::Computer.new('X'), Players::Human.new('O'), Board.new).play
+  end
+  play_again?
+end
+
+def play_again?
+  puts "Would you like to play again? Y/N"
+    input = gets.strip
+    if input == 'y'
+      start
+    else
+      puts "Good game Goodbye"
+    end
+  end
+
+start
```

Comment:
Prefer to encapsulate CLI logic into a class, like what you did for your CLI portfolio project. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.